scRNA-seq/05-clustering_markers_scRNA.Rmd

```diff
@@ -249,20 +249,25 @@ purrr::map(
 
 This returns a list of data frames, which isn't quite what we want.
 
-There is no built-in function that will give us just the first few _row names_, so we will have to define one, and `purrr` gives us a nice shorthand for doing that with `~` syntax.
-In this syntax we type a `~` followed by an expression (R code) that uses `.x` as a placeholder for a single element of the list.
-`purrr::map()` will then apply the expression to each element of the list, and return the results as a new list.
+There is no built-in function that will give us just the first few _row names_, so we will have to define one.
+As of version 4.1, R introduced a new approach to defining _anonymous functions_ - that is, functions you can quickly define "on-the-fly" without formally assigning them to a function name.
+They are handy when you need to do a very short task that requires a function, but it isn't really a function you need beyond this context.
+This new anonymous syntax looks like this: `\(x)...` (or for slightly longer code, use curly braces as in `\(x) {...}`).
+This defines a function that takes one argument, `x`, with `...` indicating where you would put the expression to calculate.
+
+`purrr::map()` will then apply the expression in our anonymous function to each element of the list, and return the results as a new list.
 
 ```{r head_markernames, live = TRUE}
 # Get the first few row names of each table with a purrr function.
 purrr::map(as.list(markers), # convert markers to a 'regular' list for purrr
-           ~ rownames(head(.x))) # our custom function.
+           \(x) rownames(head(x))) # our custom function.
```

Comment:
Maybe... I do think we should reverse the order of the two functions applied, but feel less certain about whether adding a pipe in the middle helps. 
```suggestion
           \(x) rownames(x) |> head() ) # our custom function.
```

Reply:
I think for this I prefer no pipe since the syntax is already busy with symbols. But definitely `head( rownames(x) )` in that order!